src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -741,10 +746,20 @@ def gen_code(self, parent):
         self.validate_global_constraints()
 
         extra_clauses = ""
+        # Find the specified clauses
+        clause_list = []
         if self._grainsize is not None:
-            extra_clauses = "grainsize({0})".format(self._grainsize)
+            clause_list.append("grainsize({0})".format(self._grainsize))
         if self._num_tasks is not None:
-            extra_clauses = "num_tasks({0})".format(self._num_tasks)
+            clause_list.append("num_tasks({0})".format(self._num_tasks))
+        if self._nogroup:
+            clause_list.append("nogroup")
+
+        # Generate the string containing the required clauses
```

Comment:
I think you can just do:

    extra_clauses = ", ".join(clause_list)

Reply:
Done